javascript-calculator/src/reducers/calculateReducer.js

```diff
@@ -0,0 +1,115 @@
+                    let value = prevState.equation;
+                    value = value.substring(0, value.length - 2)
+                    value += action.payload
+                    equation = value
+                } else if (prevState.equation.includes("*-") === true && action.payload !== "-") {
+                    let value = prevState.equation;
+                    value = value.substring(0, value.length - 2)
+                    value += action.payload
+                    equation = value
+                }
+                else if (prevState.equation.match(/[+/ *]$/) !== null && action.payload === "-") {
+                    let value = prevState.equation + action.payload
+                    equation = value
+                } else if (prevState.equation.match(/[+]/) !== null && action.payload !== "-") {
+                    let value = prevState.equation;
+                    value = value.substring(0, value.length - 1)
+                    value += action.payload
+                    equation = value
+                } else if (prevState.equation.match(/-/) !== null && action.payload !== "-") {
+                    let value = prevState.equation;
+                    value = value.substring(0, value.length - 1)
+                    value += action.payload
+                    equation = value
+                }
+            }
+            return { prevState, display, equation }
+        case DECIMAL_INPUT:
+            if (prevState.equation === "" || prevState.equation.includes("=")) {
+                let value = "0."
+                equation = value
+                display = value
+            } else if (prevState.equation.match(/[+\- */]$/)) {
+                let value = "0."
+                display = value
+                equation = prevState.equation + value
+            } else if (!prevState.display.includes(".")) {
+                equation = prevState.equation + action.payload
+                display = prevState.display + action.payload
+            }
+            return { prevState, display, equation }
+        case CALCULATE:
+            if (prevState.equation.includes("=")) {
+                const value = `${prevState.display} = ${prevState.display}`;
+                equation = value
+            } else if (prevState.equation !== ""
+                && prevState.equation.match(/[+\-*/]/) != null
+                && prevState.equation.match(/[+\-*/]$/) == null) {
+                const display = Number.isInteger(eval(prevState.equation)) ? eval(prevState.equation)
+                    : parseFloat(eval(prevState.equation).toFixed(5));
+                let value = prevState.equation;
+                value += ` = ${display}`;
+                return {
+                    display: String(display),
+                    equation: value
+                };
+            }
+            break;
+        case CLEAR:
+            return {
+                display: '0',
+                equation: ""
+            }
```

Comment:
Use object spread operator proposal to write `return { ...prevState, display: '0', equation: "" }`